pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -112,12 +116,34 @@ func NewController(client kube.Client, meshWatcher mesh.Holder,
 		controllers.WithReconciler(c.onEvent),
 		controllers.WithMaxAttempts(5))
 	c.ingress.AddEventHandler(controllers.ObjectHandler(c.queue.AddObject))
+
+	// We watch service changes to detect service port number change to trigger
+	// re-convert ingress to new-vs.
+	c.serviceQueue = controllers.NewQueue("service",
+		controllers.WithGenericReconciler(c.onServiceEvent),
+		controllers.WithMaxAttempts(5))
+	c.services.AddEventHandler(controllers.FromEventHandler(func(o controllers.Event) {
+		c.serviceQueue.Add(o)
```

Comment:
I did suggest filter out unneccessary service event, better  compare the old and new service

Reply:
yes. It has been implemented in    onServiceEvent. Here just enqueue service event. We will do filter and trigger vs event in the onServiceEvent.